packages/state-transition/src/block/processConsolidationRequest.ts

```diff
@@ -70,4 +79,56 @@ export function processConsolidationRequest(
     targetIndex,
   });
   state.pendingConsolidations.push(pendingConsolidation);
+
+  // Churn any target excess active balance of target and raise its max
+  if (hasEth1WithdrawalCredential(targetValidator.withdrawalCredentials)) {
+    switchToCompoundingValidator(state, targetIndex);
+  }
+}
+
+/**
+ * Determine if we should set consolidation target validator to compounding credential
+ */
+function isValidSwitchToCompoundRequest(
+  state: CachedBeaconStateElectra,
+  consolidationRequest: electra.ConsolidationRequest
+): boolean {
+  const {sourcePubkey, targetPubkey, sourceAddress} = consolidationRequest;
+  const sourceIndex = state.epochCtx.getValidatorIndex(sourcePubkey);
```

Comment:
we ciould save the index lookup if we compare the pubkeys, not sure if it's worth though but would match more closely to the spec

Reply:
> we could save the index lookup if we compare the pubkeys first, not sure if it's worth though but would match more closely to the spec

Let's do the optimization later. This part of spec is sub-optimized and will very likely to be cleaned up in alpha 9 or 10. We can optimize when the spec gets refactored.